packages/legacy/core/App/utils/helpers.ts

```diff
@@ -969,25 +948,108 @@ interface QueryParams {
+    await agent.receiveMessage(message)
+    return message
+  }
+}
+
+/**
+ * Checks if a query param object has a param Credo does not yet support but we are attempting to
+ * @param query a query object
+ * @returns the acceptable beta query param or false
  */
-const hasValidQueryParam = (query: QueryParams) => {
-  return query['d_m'] ?? query['c_i'] ?? query['oob'] ?? query['_oob'] ?? query['_url'] ?? false
+const queryHasBetaParam = (query: QueryParams) => {
+  return query['_oob'] ?? query['_url'] ?? false
+}
+
+/**
+ * Using built-in Credo methods, receive a message from a scan or deeplink and navigate accordingly
+ * @param uri a URI either containing a base64 encoded connection invite in its query parameters or a redirect URL itself
+ * @param agent an Agent instance
+ * @param navigation a navigation object from either the Scan screen, Home screen, or PasteUrl screen
+ * @param implicitInvitations a boolean to determine if implicit invitation behavior should be used
+ * @param reuseConnection a boolean to determine if connection reuse should be allowed
+ * @throws different types of CredoError if unsuccessful, what type depends on the reason for failure
+ */
+const primaryConnectStrategy = async (
+  uri: string,
+  agent: Agent | undefined,
+  navigation: any,
+  implicitInvitations: boolean = false,
+  reuseConnection: boolean = false
+) => {
+  const receivedInvitation = await connectFromInvitation(uri, agent, implicitInvitations, reuseConnection)
+  if (receivedInvitation?.connectionRecord?.id) {
+    // connection-based
+    navigation.navigate(Stacks.ConnectionStack as any, {
+      screen: Screens.Connection,
+      params: { connectionId: receivedInvitation.connectionRecord.id },
+    })
+  } else {
+    // connectionless
+    navigation.navigate(Stacks.ConnectionStack as any, {
+      screen: Screens.Connection,
+      params: { threadId: receivedInvitation?.outOfBandRecord.outOfBandInvitation.threadId },
+    })
+  }
+}
+
+/**
+ * Custom handling of beta query params to receive a message from a scan or deeplink and navigate accordingly
+ * @param uri a URI containing with base64 encoded query parameter
+ * @param agent an Agent instance
+ * @param navigation a navigation object from either the Scan screen, Home screen, or PasteUrl screen
+ * @param isDeepLink a boolean to communicate where the uri is coming from
+ */
+const betaConnectStrategy = async (uri: string, agent: Agent, navigation: any, isDeepLink: boolean) => {
+  const queryParams = parseUrl(uri)?.query
+
+  // if there's a valid query param, try unpacking and receiving the message
+  if (queryParams && queryHasBetaParam(queryParams)) {
+    const message = await getOobFromBetaQueryParam(queryParams, agent)
+    navigation.navigate(Stacks.ConnectionStack as any, {
+      screen: Screens.Connection,
+      params: { threadId: message['@id'] },
+    })
+
+    // if there's no valid query param and it's not a deeplink, throw an error
+  } else if (!isDeepLink) {
```

Comment:
Could this be a guard at the top. Might be more readable.
```
if (!isDeepLink || !(queryParams && queryHasBetaParam(queryParams)) {
   throw...
}
```

Reply:
done (with an `&&`)